### PR TITLE
Extend `color` for tuple of RGB components

### PR DESCRIPTION
**Pull request by t-bltg**
Allow passing arbitrary colors, e.g.:

```julia
using UnicodePlots

lineplot([1, 2], color=:black)
lineplot([1, 2], color=(0, 100, 200))
lineplot([1, 2], color=255)
```

Needed for `Plots.jl`, since we use `RGB` values internally.
Uses [`Crayon`](https://github.com/KristofferC/Crayons.jl) for color conversion: I think this is appropriate, since `Crayon` is already a `UnicodePlots` dependency used in Heatmaps, and its `ANSIColor` type is coherent with terminal color output.

With https://github.com/Evizero/UnicodePlots.jl/pull/136 applied, all tests pass with newly generated images:

```julia
Test Summary: | Pass  Total
tst_common.jl |  106    106
Test Summary:   | Pass  Total
tst_graphics.jl |   79     79
Test Summary: | Pass  Total
tst_canvas.jl |  140    140
Test Summary: | Pass  Total
tst_plot.jl   |   64     64
Test Summary:  | Pass  Total
tst_barplot.jl |   32     32
Test Summary:    | Pass  Total
tst_histogram.jl |   19     19
Test Summary:      | Pass  Total
tst_scatterplot.jl |   45     45
Test Summary:   | Pass  Total
tst_lineplot.jl |   79     79
Test Summary: | Pass  Total
tst_spy.jl    |   16     16
Test Summary:  | Pass  Total
tst_boxplot.jl |   20     20
Test Summary:  | Pass  Total
tst_heatmap.jl |   75     75
```

Fix https://github.com/Evizero/UnicodePlots.jl/issues/20.
Fix https://github.com/Evizero/UnicodePlots.jl/issues/58.

### CODE REVIEW

File: src/common.jl [hunk 1]
```diff
@@ -166,25 +167,38 @@ bordermap[:dashed] = border_dashed
...
+    printstyled(io, string(args...); color = col)
 end
-color_encode[:white] = 0b111
-color_decode[0b111]  = :white
 
-function print_color(color::UInt8, io::IO, args...)
-    col = color in keys(color_decode) ? color_decode[color] : Int(color)
-    str = string(args...)
-    printstyled(io, str; color = col)
+function crayon_256_color(color::UserColorType)::ColorType
```
Comment (t-bltg) [inline src/common.jl:186]: This seems a bit hacky, should I PR to `Crayons.jl` instead ?

**Comment by t-bltg** [comment]
The CI failures are related to having different ansicolor codes, even if visually the images are correct.

File: src/common.jl (hunk 1 above)
Comment (johnnychen94) [reply src/common.jl:186]: Proposing a stable API in Crayons sounds okay to me but I know very little about this. I don't know if it's useful, there are some discussions on https://github.com/JuliaGraphics/Colors.jl/issues/473

Comment (t-bltg) [reply src/common.jl:186]: Thanks for the link. What we need in `UnicodePlots` is something generic working on all terminals.

@KristofferC, do you have any advice for this ?

**Review by johnnychen94 — APPROVED** [review]

File: src/common.jl [hunk 2]
```diff
@@ -20,7 +20,8 @@ const DOC_PLOT_PARAMS = """
 
 - **`color`** : Color of the drawing.
   Can be any of `:green`, `:blue`, `:red`, `:yellow`, `:cyan`,
-  `:magenta`, `:white`, `:normal`
+  `:magenta`, `:white`, `:normal`, an integer in the range [0; 255],
+  or a tuple of three integers (RGB components).
```
Comment (johnnychen94) [inline src/common.jl:24]:
```suggestion
  `:magenta`, `:white`, `:normal`, an integer in the range `0`-`255` for Color256 palette,
  or a tuple of three integers as RGB components.
```

**Review by johnnychen94 — COMMENTED** [review]
Sorry for the back and forth. I just noticed that this PR doesn't have associated test cases added. Can you also add some tests for the new keyword types?

Currently, when I try `barplot(["B", "A"], [2, 1]; color=(255, 0, 0))` it's broken. Looks like something is missing here?

**Comment by t-bltg** [comment]
> I just noticed that this PR doesn't have associated test cases added

Good point, will do.

> Can you also add some tests for the new keyword types?

I don't understand, could you precise which keyword types you're talking about ?

> Currently, when I try barplot(["B", "A"], [2, 1]; color=(255, 0, 0)) it's broken.

Good catch ! Will fix.

**Comment by johnnychen94** [comment]
>>  Can you also add some tests for the new keyword types?
>
> I don't understand, could you precise which keyword types you're talking about ?

Sorry for the inaccuracy, I meant to test `NTuple{3, Integer}` types for `color` keyword.

**Comment by t-bltg** [comment]
> Sorry for the inaccuracy, I meant to test `NTuple{3, Integer}` types for `color` keyword.

Oh ok, yes :+1:

**Comment by t-bltg** [comment]
Damn, regression in color mixing:

reg

**Comment by t-bltg** [comment]
I've inserted some colors tests covering `NTuple{3, Integer}` and `Integer` in `tst_canvas`, `tst_barplot`, `tst_boxplot`, `tst_histogram`.

We now have 3 colors type: `UserColorType` (broad), `ColorType` for `UnicodePlots` canvas, and finally `JuliaColorType` for `printstyled`.

After visual inspection against master using:
```bash
root="$HOME/Downloads/UnicodePlots.jl"  # fork

for f in $(find $root/test/references -name '*.txt'); do
 echo; echo
 echo "== local($f) =="
 cat $f
 echo
 url="https://raw.githubusercontent.com/Evizero/UnicodePlots.jl/master/${f#$root/}"
 echo "== remote($url) =="
 wget -q -O - $url
done
```
, I think this PR can be reviewed again.

**Review by johnnychen94 — APPROVED** [review]
Sorry, I didn't look very carefully but the test looks good to me. Does a minor version bump sound good to you? @t-bltg

**Comment by t-bltg** [comment]
> Sorry, I didn't look very carefully but the test looks good to me. Since a lot of references are changed so I assume there will be some implicit behavior differences, I'd like to bump a minor version to indicate some "breaking" changes, does it sound good to you? @t-bltg

Yes, I'm good with the minor bump. I will apply the changes in `Plots.jl` accordingly after the release.

**Comment by johnnychen94** [comment]
It's actually a major version bump 😄  v2.0.0

**Pull request merged**